src/utilities/replaceASTVariables.d.ts

```diff
@@ -0,0 +1,16 @@
+import { Maybe } from '../jsutils/Maybe';
+import { ObjMap } from '../jsutils/ObjMap';
+
+import { ValueNode, ConstValueNode } from '../language/ast';
+
+/**
+ * Replaces any Variables found within an AST Value literal with literals
+ * supplied from a map of variable values, returning a constant value.
+ *
+ * Used primarily to ensure only complete constant values are used during input
+ * coercion of custom scalars which accept complex literals.
+ */
+export function replaceASTVariables(
+  valueNode: ValueNode,
+  variables: Maybe<ObjMap<unknown>>,
```

Comment:
just to be clear here: these are the raw (uncoerced variables) right?

Reply:
We are calling `valueToLiteral` [here](https://github.com/graphql/graphql-js/pull/3065/files#diff-39c231e63679a9d86e346550424c6deeba68d4b4685b0cf044ec5d3d0bb5314cR22) on the variables which means from my understanding it must be raw variables, as `valueToLiteral` is external input value (aka raw variables) to Literal.